sympy/discrete/tests/test_convolution.py

```diff
@@ -23,20 +24,28 @@ def test_convolution():
     # should be a divisor of 2**k
     p = 7*17*2**23 + 1
     q = 19*2**10 + 1
+
+    # ntt
     assert convolution(d, b, ntt=True, prime=q) == convolution_ntt(b, d, prime=q)
     assert convolution(c, b, prime=p) == convolution_ntt(b, c, prime=p)
     assert convolution(d, c, prime=p, ntt=True) == convolution_ntt(c, d, prime=p)
-
     raises(TypeError, lambda: convolution(b, d, ntt=True))
     raises(TypeError, lambda: convolution(b, d, ntt=True, cycle=0))
     raises(TypeError, lambda: convolution(b, d, dps=5, prime=q))
     raises(TypeError, lambda: convolution(b, d, dps=6, ntt=True, prime=q))
     raises(TypeError, lambda: convolution(b, d, fft=True, dps=7, ntt=True, prime=q))
-
     # ntt is a specialized variant of fft, TypeError should not be raised
     assert convolution(b, d, fft=True, ntt=True, prime=q) == \
             convolution_ntt(b, d, prime=q)
 
+    # fwht
+    assert convolution(a, b, dyadic=True) == convolution_fwht(a, b)
+    assert convolution(a, b, dyadic=False) == convolution(a, b)
+    raises(TypeError, lambda: convolution(b, d, fft=True, dps=2, dyadic=True))
+    raises(TypeError, lambda: convolution(b, d, ntt=True, prime=p, dyadic=True))
+    # fwht is a specialized variant of fft, TypeError should not be raised
+    assert convolution(b, d, fft=True, dyadic=True) == convolution_fwht(b, d)
```

Comment:
This might be confusing to some users. The implementations `fft` and `fwht` are similar, but they are applied in different type of convolutions (linear and dyadic). I am not sure that they should be accepted together.